istioctl/cmd/dashboard.go

```diff
@@ -381,3 +381,23 @@ func dashboard() *cobra.Command {
 
 	return dashboardCmd
 }
+
+// preferredPortForwarder tries to use the specified local port, but falls back to a dynamic port
+func preferredPortForwarder(client kubernetes.ExecClient, podName string, ns string, localPort int, podPort int, writer io.Writer) (*kubernetes.PortForward, error) { // nolint: lll
+	if !isPortAvailable(localPort) {
```

Comment:
There is a small potential for a race here between checking the port and using it. You could try building the port forwarder with the local port first and then fallback.